src/Microsoft.Management.Deployment/PackageManager.cpp

```diff
@@ -1197,5 +1309,91 @@ namespace winrt::Microsoft::Management::Deployment::implementation
             true /*canCancelQueueItem*/, nullptr /*queueItem*/, package, options, std::move(callerProcessInfoString));
     }
 
+    winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Microsoft::Management::Deployment::AddPackageCatalogResult, double> PackageManager::AddPackageCatalogAsync(winrt::Microsoft::Management::Deployment::AddPackageCatalogOptions options)
+    {
+        LogStartupIfApplicable();
+
+        // options must be set.
+        THROW_HR_IF_NULL(E_INVALIDARG, options);

```

Comment:
```suggestion
        THROW_HR_IF_NULL(E_POINTER, options);
```
From the same guidelines document I sent earlier, this is the correct HR for null parameters that should not be null.

Reply:
For the sake of discussion, I've noticed that the E_POINTER error code translates to NullReferenceException when it passes through the project layer, but it should be ArgumentNullException.

According to the .NET MSDN documentation for both [NullReferenceException](https://learn.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=net-8.0) and [ArgumentNullException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception?view=net-8.0&form=MG0AV3)
both exceptions use the same HRESULT value 0x80004003 but in different contexts. It seems that the winrt projection layer might not be translating this correctly, and it should be ArgumentNullException instead of NullReferenceException.

Reply:
CsWinRT is responsible for translating the ABI layer HRESULT into an appropriate projection in C#.  There isn't really a way to know the difference between the two cases though, so I assume they chose the more generic (and still true) `NullReferenceException`.